Automattic/wp-calypso
# composite-checkout: Replace item_original_monthly with item_subtotal_monthly

#### Changes proposed in this Pull Request

This is a follow-up to https://github.com/Automattic/wp-calypso/pull/42674 which used the monthly price breakdown from the shopping cart endpoint to display the monthly cost in checkout. However, that PR used each product's original cost, which could be misleading if that cost is discounted.

This PR replaces it with the item's monthly subtotal cost, which should include all discounts.

Requires D44773-code

Fixes https://github.com/Automattic/wp-calypso/issues/43123

Before:

![Screen Shot 2020-06-09 at 4 15 13 PM](https://user-images.githubusercontent.com/2036909/84196884-e860bd80-aa6e-11ea-862f-5129505193f4.png)

After:

![Screen Shot 2020-06-10 at 8 43 31 PM](https://user-images.githubusercontent.com/2036909/84332489-10c5e600-ab5b-11ea-9e61-6665326cb6a5.png)

#### Testing instructions

- Apply D44773-code and sandbox the store
- Make sure your user has an existing plan (eg: Personal).
- Add a higher-level plan to your cart (eg: Premium).
- Visit composite checkout and be sure that the monthly price display agrees with the displayed price.
